### PR TITLE
[coll] Expose configuration.

### PR DESCRIPTION
**Pull request by trivialfis**
This PR adds a configuration parameter to the dask interface for specifying communicator configurations.

Stacked on top of https://github.com/dmlc/xgboost/pull/10973 .

~todos~
- [ ] ~spark~

### CODE REVIEW

File: python-package/xgboost/collective.py [hunk 1]
```diff
@@ -15,7 +16,41 @@
 LOGGER = logging.getLogger("[xgboost.collective]")
 
 
-def init(**args: Any) -> None:
+_ArgVals: TypeAlias = Optional[Union[int, str]]
+_Args: TypeAlias = Dict[str, _ArgVals]
+
+
+@dataclass
+class Config:
```
Comment (wbo4958) [inline python-package/xgboost/collective.py:24]: Could we rename it make it more readable?

Comment (wbo4958) [reply python-package/xgboost/collective.py:24]: And could we add json support in it?

Comment (trivialfis) [reply python-package/xgboost/collective.py:24]: `xgboost.collective.Config` seems readable to me.

**Comment by trivialfis** [comment]
@wbo4958 Might be related https://github.com/dmlc/xgboost/issues/10986 .

File: python-package/xgboost/collective.py (hunk 1 above)
Comment (trivialfis) [reply python-package/xgboost/collective.py:24]: Removed the converters. We can use the utilities from `dataclasses`, see the collective test for an exmaple.

**Comment by trivialfis** [comment]
cc @rongou .

**Review by rongou — APPROVED** [review]

**Pull request merged**